File: code/data_storage/java/legos/quarkus-app/src/main/java/com/ggeoffre/data/SensorDataAccess.java
// SPDX-License-Identifier: GPL-3.0-or-later
// Copyright (C) 2025-2026 ggeoffre, LLC

package com.ggeoffre.data;

import java.util.List;

public interface SensorDataAccess {
    void logSensorData(String jsonData);
    List<String> fetchSensorData();
    void purgeSensorData();
}

File: code/data_storage/go/legos/echo-app/data/sensor_data_access.go
// SPDX-License-Identifier: GPL-3.0-or-later
// Copyright (C) 2025-2026 ggeoffre, LLC

package data

import "fmt"

// SensorDataAccess defines the interface for data access operations
type SensorDataAccess interface {
    LogSensorData(jsonData string) error
    FetchSensorData() ([]string, error)
    PurgeSensorData() error
}

// Helper function to handle unsupported data access types
func UnsupportedDataAccess(dataAccessType string) {
    panic(fmt.Sprintf("Unsupported DATA_ACCESS type: %s", dataAccessType))
}
